### PR TITLE
chore: migrate Shape component to be tailwind-compatible

### PR DESCRIPTION
**Pull request by anucreative**
## DESCRIPTION

Migrate Shape component to be Tailwind-compatible

Notes:
- Uses inline styles so we can pass a height and width in px

## SCREENSHOTS / SCREEN RECORDINGS

image

## COMPATIBILITY

- [ ] Tested on Safari (desktop)
- [x] Tested on Chrome (desktop)
- [ ] Tested on Firefox (desktop)
- [ ] Tested on mobile device sizes
- [ ] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [x] Thoroughly tested in local environment
- [ ] Added tests for all new features
- [ ] Added tests that considered edge cases

### CODE REVIEW

File: lib/src/components/Shape/index.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { getMax } from './utils'
+
+const cx = classNames(styles)
+
+export interface ShapeOptions {
+  borderRadius?: string
+  className?: string
+  h?: string
+  shape?: 'circle' | 'square'
+  w?: string
```
Comment (theo-mesnil) [inline lib/src/components/Shape/index.tsx:15]: I think this can be width or height to not have same value as styled

File: lib/src/components/Shape/index.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    let style = {}
+
+    if (borderRadius && shape !== 'circle') {
+      style = { ...style, borderRadius: `var(--radius-${borderRadius})` }
+    }
+
+    if (maxWidthHeight) {
+      style = { ...style, height: maxWidthHeight, width: maxWidthHeight }
+    }
```
Comment (theo-mesnil) [inline lib/src/components/Shape/index.tsx]: Can this be class ? It's better to avoid inline style

File: lib/src/components/Shape/shape.module.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    }
+  }
+
+  .shape-circle {
+    border-radius: 50%;
+  }
+
+  .shape-square {
+    border-radius: 0;
+  }
```
Comment (theo-mesnil) [inline lib/src/components/Shape/shape.module.scss]: Please do same as button example with variable

File: lib/src/components/Shape/index.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+export interface ShapeOptions {
+  borderRadius?: string
+  className?: string
+  h?: string
+  shape?: 'circle' | 'square'
+  w?: string
+}
+
+export type ShapeProps = CreateWuiProps<'div', ShapeOptions>
```
Comment (theo-mesnil) [inline lib/src/components/Shape/index.tsx:18]: Is the old way to do, do not use @old project on new components

File: lib/src/components/Shape/tests/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        {content}
+      </Shape>
+    )
+
+    expect(screen.getByText(content)).toBeVisible()
+  })
+
+  it('should render correctly with borderRadius', () => {
+    render(
+      <Shape borderRadius="lg" dataTestId="shape" h="100px" w="100px">
```
Comment (theo-mesnil) [inline lib/src/components/Shape/tests/index.test.tsx:21]:
```suggestion
      <Shape borderRadius="lg" data-testid="shape" h="100px" w="100px">
```

File: lib/src/components/Shape/index.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    const maxWidthHeight = (w || h) && getMax(w, h)
+
+    let style = {}
+
+    if (borderRadius && shape !== 'circle') {
+      style = { ...style, borderRadius: `var(--radius-${borderRadius})` }
+    }
+
+    if (maxWidthHeight) {
+      style = { ...style, height: maxWidthHeight, width: maxWidthHeight }
```
Comment (theo-mesnil) [inline lib/src/components/Shape/index.tsx:27]: We need to find a solution to not use style

File: lib/src/components/Shape/index.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import { classNames } from '@/utils'
+import type { CreateWuiProps } from '@old/System'
+import { forwardRef } from '@old/System'
```
Comment (theo-mesnil) [inline lib/src/components/Shape/index.tsx:3]:
```suggestion
import { forwardRef } from 'react'
```

File: website/build-app/examples.js [hunk 8]
```diff
@@ -3,18 +3,66 @@
 This file is auto-generate with yarn watch command, do not change it directly!
 **/
 
-import dynamic from "next/dynamic";
+import dynamic from 'next/dynamic'
```
Comment (theo-mesnil) [inline website/build-app/examples.js:5]: do no save this file, is auto-generate

File: lib/src/components/Shape/index.tsx (hunk 6 above)
Comment (Leiksa) [reply lib/src/components/Shape/index.tsx:27]: IMO, we could provide a "preset" sizes / arbitrary sizes, that could cover some use cases, but in general the dev could just use tailwind height/width utility classes as `className`

File: lib/src/components/Shape/index.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  ({ borderRadius, children, className, dataTestId, h, shape = 'square', w }, ref) => {
+    const maxWidthHeight = (w || h) && getMax(w, h)
+
+    let style = {}
+
+    if (maxWidthHeight) {
+      style = { ...style, height: maxWidthHeight, width: maxWidthHeight }
+    }
+
+    const radius = borderRadius && shape !== 'circle'
```
Comment (Leiksa) [inline lib/src/components/Shape/index.tsx:30]: Same here for the border-radius, we could just use in the implementation the [tailwind radius](https://tailwindcss.com/docs/border-radius) utility classes

**Comment by anucreative** [comment]
Shape will be deprecated
